services/user/CommonSys/common/packages/wasm-loader/src/importables.js

```diff
@@ -1,21 +1,164 @@
-export async function funccall(params) {
-  return "funcCall";
+const UNINITIALIZED = "uninitialized";
+
+let callerService = UNINITIALIZED;
+let callerOrigin = UNINITIALIZED;
+let myService = UNINITIALIZED;
+let myOrigin = UNINITIALIZED;
+
+function pluginId(namespace, pkg) {
+    return {
+        service: namespace,
+        plugin: pkg
+    };
 }
 
-export function getLoggedInUser() {
-  return "alice";
+function commonErr(val) {
+    return {
+        code: 0,
+        producer: pluginId("common", "plugin"),
+        message: val
+    };
 }
 
-export async function prnt(string) {
-  console.log(`from imported code:`, string);
+function send(req) {
+    try {
+        const xhr = new XMLHttpRequest();
+        xhr.open(req.method.toString(), req.uri, false);
+        const requestHeaders = new Headers(req.headers);
+        for (let [name, value] of requestHeaders.entries()) {
+            if (name !== "user-agent" && name !== "host") {
+                xhr.setRequestHeader(name, value);
+            }
+        }
+        xhr.send(req.body && req.body.length > 0 ? req.body : null);
+        if (xhr.status === 500) {
+            throw commonErr(`${xhr.response}`);
+        }
+        const body = xhr.response || undefined;
+        const headers = [];
+        xhr.getAllResponseHeaders()
+            .trim()
+            .split(/[\r\n]+/)
+            .forEach((line) => {
+                var parts = line.split(": ");
+                var key = parts.shift();
+                var value = parts.join(": ");
+                headers.push([key, value]);
+            });
+        return {
+            status: xhr.status,
+            headers,
+            body
+        };
+    } catch (err) {
+        throw new Error(err.message);
+    }
+}
 
-  const toCall = {
-    service: "idunno",
-    method: "stilldontknow",
-    params: "yesbuddeh",
-  };
+function postGraphQL(url, graphql) {
+    return send({
+        uri: url,
+        method: "POST",
+        headers: {
+            "Content-Type": "application/graphql"
+        },
+        body: graphql
+    });
 }
 
-export async function addToTx(params) {
-  console.log("addedToTx", params);
+function isValidAction(s) {
+    return /^[a-zA-Z0-9]+$/.test(s);
 }
+
+// This will be statically replaced
+const actions = [];
+
+function alreadyAdded(action, args) {
+    let dupIndex = actions.findIndex((a) => a.action === action);
+    if (dupIndex !== -1) {
+        if (args !== actions[dupIndex].args) {
+            throw new Error("Mismatched args");
+        }
+        return true;
+    }
+    return false;
+}
+
+export const server = {
+    // add-action-to-transaction: func(service: string, action: string, args: list<u8>) -> result<_, string>;
+    addActionToTransaction(action, args) {
+        if (myService === UNINITIALIZED)
```

Comment:
I'd add the brackets here for consistency. And there are a couple opportunities in this function to drop some `else`'s too...but I'm getting pretty nitpicky at this point haha. Same goes for the rest of this file.